modules/custom/d_p_side_by_side/d_p_side_by_side.module

```diff
@@ -26,3 +26,15 @@ function d_p_side_by_side_entity_bundle_field_info_alter(&$fields, EntityTypeInt
     ]);
   }
 }
+
+/**
+ * Implements hook_preprocess().
+ *
+ * Check if side by side contains non-empty header.
+ */
+function template_preprocess_paragraph__d_p_side_by_side(&$variables, $hook) {
```

Comment:
I would put this preprocess into `d_p_side_by_side.module` as `d_p_side_by_side_preprocess_paragraph__d_p_side_by_side`